pkg/kubelet/volume/gce_pd/gce_pd.go

```diff
@@ -263,7 +263,14 @@ func (pd *gcePersistentDisk) TearDown() error {
 	// If refCount is 1, then all bind mounts have been removed, and the
 	// remaining reference is the global mount. It is safe to detach.
 	if refCount == 1 {
-		if err := pd.manager.DetachDisk(pd, devicePath); err != nil {
+		// pd.pdName is not initially set for volume-cleaners, so set it here.
+		var err error
+		globalMountPath := path.Join(pd.plugin.host.GetPluginDir(gcePersistentDiskPluginName), "mounts")
+		pd.pdName, err = getPDName(globalMountPath, pd.mounter, devicePath)
```

Comment:
Good call.

The device path is no longer a reliable way to retrieve the name of the PD, because these mounts are now showing up in the mount table with their canonical path (e.g. /dev/sdb) instead of their google-friendly symlinked path (/dev/disk/by-id/google-mypd)

I changed that function to return a list of other mount paths. If there is only one left, it is the global mount path, and we can grab the PD name off of the end (because we define that path).